07_safety/juvenile-arrests/juvenile-arrests-county-all.Rmd

```diff
@@ -0,0 +1,971 @@
+
+
+## 3. Read in and clean NIBRS arrest data
+This section extracts the necessary files from each zip file downloaded in the previous step and reads the files into the R environment. Each zip file contains data files for 1991-2022. However, we only want 2021-2022. Note that future updates of this metric may want to include historical NIBRS data back to 2014 and should update this section to include all years.
+
+### 3.1 Read in NIBRS data
+
+#### 3.1a Extract files for 2021-2022 from the zip files for both segments
+Extract each file for the years 2021-2022 from the Group A `arrestee` segment and from the Group B `arrest_report` segment.
+```{r}
+# Specify range of years
+start_year <- 2021
+end_year <- 2022
+
+# Create a vector of years between start_year and end_year
+my_years <- seq(start_year, end_year)
+
+# Extract years 2014-2022 for the arrestee segment
+for (year in my_years) {
+  unzip(
+  zipfile = here::here("07_safety", "juvenile-arrests", "data", "nibrs_1991_2022_arrestee_segment_rds.zip"),
+  files = paste0("nibrs_arrestee_segment_", year, ".rds"),
+  exdir = here::here("07_safety", "juvenile-arrests", "data", "nibrs_1991_2022_arrestee_segment_rds")
+)
+}
+
+# Extract years 2014-2022 for the Group B arrest report segment
+for (year in my_years) {
+  unzip(
+  zipfile = here::here("07_safety", "juvenile-arrests", "data", "nibrs_1991_2022_group_b_arrest_report_segment_rds.zip"),
+  files = paste0("nibrs_group_b_arrest_report_segment_", year, ".rds"),
+  exdir = here::here("07_safety", "juvenile-arrests", "data", "nibrs_1991_2022_group_b_arrest_report_segment_rds")
+)
+}
+
+# # Remove zip files
+# unlink(here::here("07_safety", "juvenile-arrests", "data", "nibrs_1991_2022_arrestee_segment_rds.zip"))
+# 
+# unlink(here::here("07_safety", "juvenile-arrests", "data", "nibrs_1991_2022_group_b_arrest_report_segment_rds.zip"))
+
+# Clean up environment
+rm(start_year, end_year, my_year)
+
+```
+
+#### 3.1b Read in and append 2021-2022 NIBRS data for both groups
+For each segment, read in the files for years 2021-2022 and append them. We also limit the appended files to relevant variables. 
+Note that when including historical data back to 2014, this section has a very long run time. The reviewer or future updates may want to consider writing a more efficient approach if possible.    
+```{r}
+# List all files in the Arrestee segment folder (Group A)
+file_list <- list.files(here::here(
+  "07_safety", "juvenile-arrests", "data", "nibrs_1991_2022_arrestee_segment_rds"),
+  pattern = "[.]rds$", full.names = TRUE)
+
+# Read in Group A files and append them
+arrests_a <- file_list %>%
+  set_names(basename) %>%
+  purrr::map(readRDS) %>%
+  list_rbind(names_to = "year") %>%
+  mutate(year = parse_number(year))
+
+# Select only necessary variables from Group A and limit observations to under age 18
+arrests_a <- arrests_a %>%
+  select(year, ori, # ORI is a unique agency identifier
```

Comment:
I'm not getting that error. I opened the raw data for the Group A files (`nibrs_arrestee_segment_2021` and `nibrs_arrestee_segment_2022`) and confirmed that they both contain a variable titled `incident_number`. 

This variable is titled `arrest_transaction_incident_num` in the Group B files which is why it is renamed in the section that cleans the Group B data.

Regardless, these variables aren't actually used in the analysis. I kept them in because they were also included in the code for the last update and I thought maybe future updates to the metric planned on using them for something. For now, I am just deleting them.